docs/content/guides/developer/app-examples/escrow-indexer-api.mdx

```diff
@@ -0,0 +1,634 @@
+---
+title: Escrow Indexing and API Service
+sidebar_label: Indexing and API Service
+---
+
+:::note Multi-Page Guide
+
+This is the second in a [three-part guide](/guides/developer/app-examples/escrow) on how to build a trustless atomic swap on Sui.
+
+:::
+
+In most cases where we want to enhance an dApp and ensure it is production ready, we need to have an indexing service (indexer) listening to the blockchain for on-chain data, shaping the data to fit our application needs, storing the transformed data into the local off-chain database so we can query them in the most efficient way. Joining hands with an indexer, we will expose an API allowing frontend to query the indexed data and update the screen as escrows are made and swaps are fulfilled.
+
+Architecturally, we will have the service doing the heavy-lifting indexing the data while the other service will expose the data through an API convention for external consumption.
```

Comment:
nit: this sentence reads a little strangely -- could you have a look and possibly reword it?